src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs

```diff
@@ -87,6 +88,21 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
 				}
 				else
 				{
+					var matrix = Matrix3x2.CreateTranslation((float)backgroundArea.Left, (float)backgroundArea.Top);
+					matrix *= TransformMatrix;
+
+					// The brush is not tied to a specific window, so we can't get the scaling of a specific XamlRoot and
+					// instead we settle for the main window's scaling which should hopefully be the same.
+					var scale = DisplayInformation.GetForCurrentViewSafe().LogicalDpi / DisplayInformation.BaseDpi;
```

Comment:
Are we sure this is the behavior on WinUI?

Reply:
What behaviour? We aren't changing anything. I'm just adding a little hack to keep images crisp under high scaling (this is due to the previous workaround that uses Lancsoz instead of letting the shader deal with scaling, which is originally because of skia's scaling algorithm being really really bad quality).

Reply:
The fact that DPI scaling is taken into consideration. NOTE: The issue was opened actually before your previous fix.

Reply:
Yes, WinUI respects the scaling. I didn't notice the date. I'm not sure why it was broken before the previous fix (maybe it was just the bad scaling algorithm), but it should be working now or at least another problem should be fixed.

@ArchieCoder Can we get your exact repro for https://github.com/unoplatform/uno/issues/16697 to see if this indeed fixes your problem?

Reply:
The exact repro was listed in the original bug: create default project then deploy. No code changed needed.

My monitor has Scale 150% for a resolution of 3840x2160.

It is still fuzzy. 

![image](https://github.com/unoplatform/uno/assets/1608424/b7b8e321-7eb3-44a7-b8ee-acf9de80b3c3)

Reply:
My bad, I missed it. I've tested with this PR branch and it seems to be better (I'm not sure if it's "perfect", but definitely a good improvement). When you say "it's still fuzzy", do you mean that you tried with this branch?

here's a before and after
![image](https://github.com/unoplatform/uno/assets/66218781/6ae4031f-a9d4-4638-aae8-e385763218b7)
![image](https://github.com/unoplatform/uno/assets/66218781/77f6ba5b-a801-430a-8a47-4ba3352abdee)

Reply:
@ramezgerges I didn't try with the branch, I don't have this setup. I can update the Uno.Sdk number or some nugets on demand. I thought there was a change since one month ago.